modules/api/src/test/java/org/apache/fluo/api/data/AsciiSequence.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.fluo.api.data;
+
+import java.nio.charset.StandardCharsets;
+
+class AsciiSequence implements CharSequence {
```

Comment:
It's my understanding that behavior related to auxiliary classes aren't very well defined in the JLS. In general, they are treated like package-private classes, but not all compilers treat them exactly the same way. As such, they aren't safe to reference outside of its source file, so the Java compiler warns about that. Auxiliary classes also violate Oracle's documented code conventions (which admittedly haven't been updated since 1999). It's better to follow the one-class-per-file standard. Alternatively, a static inner-class could be an appropriate alternative.